### PR TITLE
8310108: Skip ReplaceCriticalClassesForSubgraphs when EnableJVMCI is specified

### PR DESCRIPTION
**Pull request by mur47x111**
EnableJVMCI appends the jdk.internal.vm.ci module and disables full module graph. Temporarily bypasses FMG-related tests with EnableJVMCI until JDK-8266329 is resolved.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8310108](https://bugs.openjdk.org/browse/JDK-8310108): Skip ReplaceCriticalClassesForSubgraphs when EnableJVMCI is specified (**Bug** - P4)

### Reviewers
 * [Doug Simon](https://openjdk.org/census#dnsimon) (@dougxc - Committer)
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14485/head:pull/14485` \
`$ git checkout pull/14485`

Update a local copy of the PR: \
`$ git checkout pull/14485` \
`$ git pull https://git.openjdk.org/jdk.git pull/14485/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14485`

View PR using the GUI difftool: \
`$ git pr show -t 14485`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14485.diff">https://git.openjdk.org/jdk/pull/14485.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14485#issuecomment-1592498948)

### CODE REVIEW

**Review by dougxc — APPROVED** [review]

File: test/hotspot/jtreg/runtime/cds/serviceability/ReplaceCriticalClassesForSubgraphs.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
  * @summary Tests how CDS works when critical library classes are replaced with JVMTI ClassFileLoadHook
  * @library /test/lib
  * @requires vm.cds.write.archived.java.heap
+ * @requires !vm.jvmci.enabled
```
Comment (dougxc) [inline test/hotspot/jtreg/runtime/cds/serviceability/ReplaceCriticalClassesForSubgraphs.java:30]: Please add a comment to https://bugs.openjdk.org/browse/JDK-8266329 to remove this guard as part of resolving the issue.

**Review by iklam — APPROVED** [review]
LGTM.

**Comment by mur47x111** [comment]
@iklam @dougxc thanks for the review!

**Pull request closed without merging**